src/renderer/builder.rs

```diff
@@ -0,0 +1,287 @@
+        self.uri_scheme_alias = uri_scheme_alias.to_string();
+        self
+    }
+
+    /// Sets source JSON URL template
+    ///
+    /// Default: "/tiles/{domain}.json"
+    #[must_use]
+    #[allow(clippy::needless_pass_by_value, reason = "false positive")]
+    pub fn with_source_template(mut self, source_template: impl ToString) -> Self {
+        self.source_template = source_template.to_string();
+        self
+    }
+    /// Sets style JSON URL template
+    ///
+    /// Default: "{path}.json"
+    #[must_use]
+    #[allow(clippy::needless_pass_by_value, reason = "false positive")]
+    pub fn with_style_template(mut self, style_template: impl ToString) -> Self {
+        self.style_template = style_template.to_string();
+        self
+    }
+
+    /// Sets sprite URL template
+    ///
+    /// Default: "/{path}/sprite{scale}.{format}"
+    #[must_use]
+    #[allow(clippy::needless_pass_by_value, reason = "false positive")]
+    pub fn with_sprites_template(mut self, sprites_template: impl ToString) -> Self {
+        self.sprites_template = sprites_template.to_string();
+        self
+    }
+
+    /// Sets glyph URL template
+    ///
+    /// Default: "/font/{fontstack}/{start}-{end}.pbf"
+    #[must_use]
+    #[allow(clippy::needless_pass_by_value, reason = "false positive")]
+    pub fn with_glyphs_template(mut self, glyphs_template: impl ToString) -> Self {
+        self.glyphs_template = glyphs_template.to_string();
+        self
+    }
+
+    /// Sets tile URL template
+    ///
+    /// Default: "/{path}"
+    #[must_use]
+    #[allow(clippy::needless_pass_by_value, reason = "false positive")]
+    pub fn with_tile_template(mut self, tile_template: impl ToString) -> Self {
+        self.tile_template = tile_template.to_string();
+        self
+    }
+
+    /// Sets API key parameter name
+    ///
+    /// Default: ""
+    #[must_use]
+    #[allow(clippy::needless_pass_by_value, reason = "false positive")]
+    pub fn with_api_key_parameter_name(mut self, api_key_parameter_name: impl ToString) -> Self {
+        self.api_key_parameter_name = api_key_parameter_name.to_string();
+        self
+    }
+
+    /// Sets API key
+    ///
+    /// Default: ""
+    #[must_use]
+    #[allow(clippy::needless_pass_by_value, reason = "false positive")]
+    pub fn with_api_key(mut self, api_key: impl ToString) -> Self {
+        self.api_key = api_key.to_string();
+        self
+    }
+
+    /// Sets whether API key is required
+    ///
+    /// Default: `false`
+    #[must_use]
+    pub fn set_requires_api_key(mut self, requires_api_key: impl Into<bool>) -> Self {
+        self.requires_api_key = requires_api_key.into();
+        self
+    }
+
+    /// Builds a static image renderer
+    #[must_use]
+    pub fn build_static_renderer(self) -> ImageRenderer<Static> {
+        // TODO: Should the width/height be passed in here, or have another `build_static_with_size` method?
```

Comment:
if the img size is a param to the renderer, you probably could pass it in every time in the `render` method, just like we do with the tile's x/y/z/dpi.  If it has to remain fixed for all images, and just change the location, may make sense to set it here

Reply:
Not sure what you or the comment is referring to.

Image size is not something that can change (?) currently unless you fully restart.
Maybe MLN has an API for this, not sure.

That seems unrelated to this change, though.